lib/services/ldap-provider.js

```diff
@@ -1,22 +1,41 @@
 'use strict';
 
-var config = require('./../../config.js'),
+var config = require('../config'),
     logger = require('../logger'),
     ldap = require('ldapjs');
 
-var searchBase = config.ldap.searchBase;
+var ldapConfig = config.get('ldap');
 
 var client = ldap.createClient({
-  url: config.ldap.url,
-  maxConnections: config.ldap.maxConnections,
+  url: ldapConfig.url,
+  maxConnections: ldapConfig.maxConnections
 });
 
 var validate = function(email, cb) {
+  //obtain email domain
+  var parts = email.split('@');
+  var emailDomain = parts[1];
+  var domain = (ldapConfig.domains[emailDomain] === undefined) ? 'default' : emailDomain;
+  var filterValue = null;
+
+  //if there is regex configured for that domain, apply it, otherwise just take email as received
+  if (ldapConfig.domains[domain].filterFieldRegEx) {
```

Comment:
I would avoid this if if we just force filterFieldRegEx to be always defined. You can use jsonschema...